src/game_inventory.cpp

```diff
@@ -92,62 +118,222 @@ int game::inv_for_unequipped( const std::string &title )
-        return item_location();
-    }
+        bool is_shown( const item &it ) const override {
+            return it.is_reloadable_with( liquid.typeId() ) || it.is_container();
+        }
 
-    return std::move( inv_s.execute() );
-}
+        bool is_enabled( const item_location &location ) const override {
+            return !is_same_container( *location.get_item() ) && get_rem_capacity( location ) > 0;
+        }
+
+        int get_rank( const item_location &location ) const override {
+            return -get_rem_capacity( location );
+        }
+
+    protected:
+        bool is_same_container( const item &it ) const {
+            return it.is_container() && &it.contents.front() == &liquid;
+        }
+
+        int get_rem_capacity( const item_location &location, std::string *err = nullptr ) const {
+            if( location.where() == item_location::type::character ) {
+                Character *character = dynamic_cast<Character *>( g->critter_at( location.position() ) );
+                if( character == nullptr ) {
+                    debugmsg( "Supplied an invalid location: no character found." );
+                    return 0;
+                }
+                return location->get_remaining_capacity_for_liquid( liquid, *character, err );
+            } else {
+                const bool allow_buckets = location.where() == item_location::type::map;
+                return location->get_remaining_capacity_for_liquid( liquid, allow_buckets, err );
+            }
+        }
+
+    private:
+        const item &liquid;
+};
 
 item *game::inv_map_for_liquid( const item &liquid, const std::string &title, int radius )
 {
-    const auto filter = [ this, &liquid ]( const item_location & location ) {
-        if( location.where() == item_location::type::character ) {
-            Character *character = dynamic_cast<Character *>( critter_at( location.position() ) );
-            if( character == nullptr ) {
-                debugmsg( "Invalid location supplied to the liquid filter: no character found." );
-                return false;
+    return inv_internal( u, liquid_inventory_preset( liquid ), title, radius,
+                         string_format( _( "You don't have a suitable container for carrying %s." ),
+                                        liquid.tname().c_str() ) ).get_item();
+}
+
+class comestible_inventory_preset: public inventory_selector_preset
+{
+    public:
+        comestible_inventory_preset( const player &p ) : p( p ) {
+            append_cell( [ this ]( const item & it ) {
+                return to_string( this->p.nutrition_for( get_food_type( it ) ) );
+            }, _( "NUTRITION" ) );
+
+            append_cell( [ this ]( const item & it ) {
+                return to_string( get_comestible( it ).quench );
+            }, _( "QUENCH" ) );
+
+            append_cell( [ this ]( const item & it ) {
+                return to_string( get_comestible( it ).fun );
+            }, _( "  FUN" ) );
+        }
```

Comment:
Translators don't like it when there are (unexplained) spaces at the start / end (or maybe anywhere) of the string. In this case, does it mean the string must be *exactly* 5 characters long? If not, why are there spaces?